scripts/build.js

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env node
+const execa = require('execa');
+const program = require('commander');
+const path = require('path');
+const remove = require('lodash/remove');
+const EventEmitter = require('events');
+const { Worker } = require('worker_threads');
+
+const myEmitter = new EventEmitter();
+
+let modules;
+let i = 0;
+
+async function retrieveDependencies(_modules) {
+  const rtn = _modules;
+  const packages = await Promise.all(
+    rtn.map(pck => execa
+      .shell(
+        `lerna list -alp --include-filtered-dependencies --json --scope="${pck.name}"`,
+      )
+      .then(({ stdout }) => {
+        const dependents = JSON.parse(stdout).filter(
+          p => p.name !== pck.name,
+        );
+        return Object.assign(pck, { dependents });
+      })),
+  );
+  return packages;
+}
+
+function unstack() {
+  console.log(`-=-=-=- #${i} -=-=-=-`);
+  console.log(
+    `TODO (${modules.todo.length})`,
+    modules.todo.map(m => m.name).join(', '),
+  );
+  console.log(
+    `DOING (${modules.doing.length})`,
+    modules.doing.map(m => m.name).join(', '),
+  );
+  console.log(
+    `DONE (${modules.done.length})`,
+    modules.done.map(m => m.name).join(', '),
+  );
+  modules.todo.map((pck) => {
+    const deps = pck.dependents.filter(
+      dep => !modules.done.find(el => el.name === dep.name),
+    );
+
+    if (deps.length === 0) {
+      const workerData = remove(modules.todo, p => p.name === pck.name)[0];
+      modules.doing.push(workerData);
+
+      const promise = program.dryRun
+        ? new Promise(resolve => resolve())
+        : new Promise((resolve, reject) => {
+          const worker = new Worker(path.join(__dirname, '/build_module.js'), {
+            workerData,
+          });
+
+          worker.on('online', () => {
+            console.log(
+              `Launching intensive build task for package ${workerData.name}`,
+            );
+          });
+          worker.on('message', (messageFromWorker) => {
+            console.log(messageFromWorker);
+          });
+          worker.on('error', reject);
+          worker.on('exit', (code) => {
+            if (code !== 0) {
+              reject(new Error(`Worker stopped with exit code ${code}`));
+            }
+            resolve();
+          });
+        });
+
+      return promise.then(() => {
+        remove(modules.doing, workerData);
+        modules.done.push(workerData);
+        myEmitter.emit('unstack');
+      });
+    }
+    return null;
+  });
+  i += 1;
+}
+
+program
+  .option('--dry-run', 'Launch the build script without creating dist')
+  .option('-p, --package [package]', 'Scope build to a specific package and its dependencies')
+  .action(() => {
+    execa
+      .shell(`lerna list -alp --json --top-o-sort ${program.package ? `--scope=${program.package} --include-filtered-dependencies` : ''}`)
```

Comment:
Regarding the package documentation, it's recommended to prefer using `execa()` instead of `execa.shell()`.

ref: https://github.com/sindresorhus/execa/tree/v1.0.0#execashellcommand-options

Reply:
I cannot find the flag `--top-o-sort` for the `lerna list` command.

Shouldn't be [`toposort`](https://github.com/lerna/lerna/tree/master/commands/list#--toposort) instead?

Reply:
Except that here, I want to use shell command, not execute a file :/

`execa()` is used to execute a file :)

Reply:
For toposort, this option is now useless in this, I removed it ;)

Reply:
Nothing blocker but we might consider bump to the latest version of execa which drop the `.shell` method in flavor of `.command` instead.

I will open a pull request to address it later on.

ref: https://github.com/sindresorhus/execa/releases/tag/v2.0.0